### PR TITLE
Add Nomenclature.md

### LINKED ISSUE #171: docs: Nomenclature
Add definitions to previous list of terms and put in dev-rel Nomenclature doc.
Some definitions may make sense in the aztec-packages glossary, but largely useful to have a place dev-rel can pin down ambiguous/cobbled definitions that surface in different places.

### PR DESCRIPTION
**Pull request by jzaki**
Closes #171

### CODE REVIEW

File: NOMENCLATURE.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+## Aztec Network
+Computers on a network that adhere to the Aztec Protocol.
+
+## Aztec Client Software
+Software run on individual computers that adhere to the Aztec protocol.
+(learn more from the Yellow Paper)
+
+## Aztec (Public) Virtual Machine
+Executes the public section of a transaction, and unconstrained functions.
```
Comment (critesjosh) [inline NOMENCLATURE.md:17]: unconstrained functions are executed by the PXE, client-side, and are just read operations

File: NOMENCLATURE.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+## Private eXecution Environment (PXE)
+Execute the private section of a transaction.
+
+## Noir Language
+A rust-like language to represent constraints for zero-knowledge circuits.
```
Comment (critesjosh) [inline NOMENCLATURE.md:23]: "Rust-like" can be confusing. it has syntax familiar to rust, but is very different. 

Consider the tagline from the Noir website

Noir is a Domain-Specific Language for SNARK proving systems developed by [Aztec Labs](https://aztec.network/). It allows you to generate complex Zero-Knowledge Programs (ZKP) by using simple and flexible syntax, requiring no previous knowledge on the underlying mathematics or cryptography.

File: NOMENCLATURE.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+Framework written in Noir to help with development of Aztec Contracts
+
+## Abstract Circuit Intermediate Representation (ACIR)
+Noir compiles to ACIR.
+
+## Abstract Circuit Virtual Machine (ACVM)
+Converts ACIR into compatible backend formats.
+
+## Noir Compiler (`nargo`)
+Prove, verify, generate solidity verifier, ...
```
Comment (critesjosh) [inline NOMENCLATURE.md:38]: maybe useful to link to the relevant docs

File: NOMENCLATURE.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+## Aztec Compiler (`aztec-nargo`)
+Interpret Noir, via ACIR.
```
Comment (critesjosh) [inline NOMENCLATURE.md:41]:
```suggestion
Interpret Noir written as Aztec contracts, via ACIR.
```

File: NOMENCLATURE.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+## Aztec Sandbox (Tool suite)
+Installed via `bash -i <(curl -s install.aztec.network)`
```
Comment (critesjosh) [inline NOMENCLATURE.md:44]: link?

File: NOMENCLATURE.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+## Aztec Client Software
+Software run on individual computers that adhere to the Aztec protocol.
+(learn more from the Yellow Paper)
+
+## Aztec (Public) Virtual Machine
+Executes the public section of a transaction, and unconstrained functions.
+
+## Private eXecution Environment (PXE)
+Execute the private section of a transaction.
```
Comment (catmcgee) [inline NOMENCLATURE.md:20]: Would be great to mention it runs locally

File: NOMENCLATURE.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+## Noir Language
+A rust-like language to represent constraints for zero-knowledge circuits.
+
+## Aztec Contract or Aztec Smart Contract
+Definitions of public and private, state and functions. Public code is execute in Protocol's AVM, private code in PXE's ACVM.
```
Comment (catmcgee) [inline NOMENCLATURE.md:26]:
```suggestion
Definitions of public and private state and functions. Public code is executed in Protocol's AVM, private code in PXE's ACVM.
```

File: NOMENCLATURE.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+## Aztec.nr (Framework/website)
+Framework written in Noir to help with development of Aztec Contracts
+
+## Abstract Circuit Intermediate Representation (ACIR)
+Noir compiles to ACIR.
```
Comment (catmcgee) [inline NOMENCLATURE.md:32]: I think we could explain more about ACIR here

File: NOMENCLATURE.md (hunk 2 above)
Comment (jzaki) [reply NOMENCLATURE.md:23]: Done.
Interestingly aztec.nr says: "Rust-like syntax"

Comment (catmcgee) [reply NOMENCLATURE.md:23]: It does have Rust-like syntax, but it's not a Rust-like language

**Review by catmcgee — APPROVED** [review]

**Pull request merged**